app/scripts/views/confirm_account_unlock.js

```diff
@@ -0,0 +1,162 @@
+  'views/mixins/back-mixin'
+],
+function (Cocktail, FormView, BaseView, Template, p, AuthErrors,
+    ResendMixin, ServiceMixin, BackMixin) {
+  var VERIFICATION_POLL_IN_MS = 4000; // 4 seconds
+
+  var View = FormView.extend({
+    template: Template,
+    className: 'confirm_account_unlock',
+
+    // used by unit tests
+    VERIFICATION_POLL_IN_MS: VERIFICATION_POLL_IN_MS,
+
+    initialize: function () {
+      var data = this.ephemeralData();
+      var accountData;
+
+      if (data) {
+        if (data.lockoutSource) {
+          this._lockoutSource = data.lockoutSource;
+        }
+
+        if (data.account) {
+          accountData = data.account;
+        }
+      }
+
+      this._account = this.user.initAccount(accountData);
+    },
+
+    getAccount: function () {
+      return this._account;
+    },
+
+    context: function () {
+      return {
+        email: this.getAccount().get('email')
+      };
+    },
+
+    events: {
+      // validateAndSubmit is used to prevent multiple concurrent submissions.
+      'click #resend': BaseView.preventDefaultThen('validateAndSubmit')
+
+    },
+
+    _bouncedEmail: function () {
+      this.ephemeralMessages.set('bouncedEmail', this.getAccount().get('email'));
+      // what to do here? Go back?
+    },
+
+    beforeRender: function () {
+      // browsing directly to the page should not be allowed.
+      var self = this;
+      return p().then(function () {
+        if (self.getAccount().isEmpty()) {
+          self.navigate('signup');
+        }
+      });
+    },
+
+    afterVisible: function () {
+      var graphic = this.$el.find('.graphic');
+      graphic.addClass('pulse');
+
+      var self = this;
+      return self.broker.persist()
+        .then(function () {
+          return self._waitForConfirmation();
+        })
+        .then(function () {
+          self.logScreenEvent('verification.success');
+          // the continuation path depends on the action that triggerred
+          // the account lockout notice. The only time the broker should
+          // be notified is if the user was trying to sign in.
+          if (/signin/.test(self._lockoutSource)) {
+            return self.broker.afterSignIn(self.getAccount());
+          }
+        })
+        .then(function (result) {
+          if (! (result && result.halt)) {
+            self.navigate('account_unlock_complete');
+          }
+        }, function (err) {
+          // The user's email may have bounced because it was invalid.
+          // Redirect them to the sign up page with an error notice.
+          if (AuthErrors.is(err, 'SIGNUP_EMAIL_BOUNCE')) {
+            self._bouncedEmail();
+          } else {
+            self.displayError(err);
+          }
+        });
+    },
+
+    _waitForConfirmation: function () {
+      var deferred = p.defer();
+      return deferred.promise;
+      /*
+      var self = this;
+      var account = self.getAccount();
+      return self.fxaClient.recoveryEmailStatus(
```

Comment:
Thinking about this more, the 3 locations where the user is shown the "please re-verify your email", we have both the user's email and password. @ckarlof is right, we can just try to sign the user in. The sign in will succeed if the user has verified their address, and fail with the `ACCOUNT_LOCKED` error if they have not.